src/main/scala/org/apache/mesos/chronos/scheduler/jobs/TaskUtils.scala

```diff
@@ -90,8 +90,8 @@ object TaskUtils {
     })
   }
 
-  def getTaskId(job: BaseJob, due: DateTime, attempt: Int = 0, arguments: String = ""): String = {
-    taskIdTemplate.format(due.getMillis, attempt, job.name, arguments)
+  def getTaskId(job: BaseJob, due: DateTime, attempt: Int = 0): String = {
+    taskIdTemplate.format(due.getMillis, attempt, job.name, job.arguments.foldRight("")((a, b) => if (b.nonEmpty) s"$a $b" else a))
   }
```

Comment:
This makes sense, but could we simplify the logic some by iterating through the list from left to right and doing a simple string concatenation. job.arguments.foldLeft("") ((a, b) => if (b.nonEmpty) a + " " + b else a))